### PR TITLE
New example that shows a combination of Async workflow with Signal and Query

### PR DESCRIPTION
**Pull request by ramkumarkb**
A new example with the following workflow:
1. Start Workflow (asynchronous)
2. Wait
3. Signal-1 => Do Activity-1
4. Wait
5. Signal-2 => Do Activity-2
6. Query the workflow

### CODE REVIEW

File: src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      }
+
+      System.out.println("Awaiting Signal2");
+      if (!signal2.get().isEmpty()) {
+        String activity2Result = activity2.composeGreeting2("Hello", "Activity-2");
+        receivedMessages.add(activity2Result);
+      }
+
+      messageQueue.addAll(receivedMessages);
+      return receivedMessages;
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java]: I would return just activity1Result + activity2Result, may be with some separator.

File: src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      signal1.complete(signal);
+    }
+
+    @Override
+    public void waitForSignal2(String signal) {
+      signal2.complete(signal);
+    }
+
+    @Override
+    public String queryResults() {
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java]: Let's remove query to keep the sample simple

File: src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    WorkflowOptions workflowOptions =
+        new WorkflowOptions.Builder()
+            .setTaskList(TASK_LIST)
+            .setExecutionStartToCloseTimeout(Duration.ofSeconds(30))
+            .build();
+    GreetingWorkflow workflow =
+        workflowClient.newWorkflowStub(GreetingWorkflow.class, workflowOptions);
+    // Start workflow asynchronously to not use another thread to query.
+    WorkflowClient.start(workflow::getGreetings);
+    Thread.sleep(1000);
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java]: This sleep is not necessary

File: src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    // After start for getGreeting returns, the workflow is guaranteed to be started.
+    // So we can send a signal to it using workflow stub.
+    // This workflow will execute different Activities based on the Signal received
+    workflow.waitForSignal1("START_ACTIVITY1");
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java]: May be rename the method as it is confusing from the caller part

File: src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    // Note that inside a workflow only WorkflowThread.sleep is allowed. Outside
+    // WorkflowThread.sleep is not allowed.
+    Thread.sleep(1500);
+    workflow.waitForSignal2("START_ACTIVITY2");
+    Thread.sleep(500);
+    // System.out.println(workflow.queryResults());
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/samples/hello/HelloAsyncWorkflowWithSignalCompletion.java]: remove the query part

**Comment by ibarrajo** [comment]
Hi @ramkumarkb ! just wanted to check if you would like to get your PR ready for review.

**Pull request closed without merging**